flax/nn/attention.py

```diff
@@ -45,7 +45,9 @@ def dot_product_attention(query,
                           dropout_rate=0.,
                           deterministic=False,
                           precision=None):
-  """Computes dot-product attention given query, key, and value.
+  """DEPRECATION WARNING:
```

Comment:
Do we not need to also call `warnings.warn(..., DeprecationWarning)`? That's how we've done deprecations, e.g. see `flax.jax_utils.pmean`.

Or maybe we've been doing it wrong the whole time? We should standardize on the right way to do deprecations in Python.

Reply:
...ah I see, you added it in flax/nn/init.py. OK maybe that's good.